scripts/delete-api-keys.js

```diff
@@ -0,0 +1,61 @@
+const axios = require('axios');
+const {homedir} = require('os');
+const {join} = require('path');
+const {config} = require('dotenv');
+config({path: join(homedir(), '.env')});
+
+// TODO: CDX-98: URL should vary in function of the target environment.
+const platformHost = 'https://platformdev.cloud.coveo.com/rest/';
+
+function wasCreatedByTheCli(testRunId = '') {
+  return (key) =>
+    key.displayName?.startsWith(`cli-${testRunId}`) &&
+    key.description === 'Generated by the Coveo CLI';
+}
+
+function isAtLeastOneDayOld(key) {
+  const today = new Date();
+  return (
+    key.createdDate && key.createdDate < today.setDate(today.getDate() - 1)
+  );
+}
+
+function authHeader(accessToken) {
+  return {
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+    },
+  };
+}
+
+async function getAllApiKeys(orgId, accessToken) {
+  const apiKeysUrl = `${platformHost}organizations/${orgId}/apikeys`;
+  const response = await axios.get(apiKeysUrl, authHeader(accessToken));
+  return response.data;
+}
+
+async function deleteApiKeys(orgId, accessToken, apiKeysToDelete) {
+  for (let i = 0; i < apiKeysToDelete.length; i++) {
+    const apiKey = apiKeysToDelete[i];
+    const apiKeyUrl = `${platformHost}organizations/${orgId}/apikeys/${apiKey.id}`;
+    console.log(`Deleting ${apiKey.displayName}`);
+    await axios.delete(apiKeyUrl, authHeader(accessToken));
+  }
+  console.log(`\nDeleted ${apiKeysToDelete.length} API keys`);
+}
+
+async function main() {
+  const testOrgId = process.env.TEST_ORG_ID;
+  const accessToken = process.env.ACCESS_TOKEN;
+  const testRunId = process.env.TEST_RUN_ID;
+  const apiKeys = await getAllApiKeys(testOrgId, accessToken);
+
+  let cliApiKeys = apiKeys.filter(wasCreatedByTheCli(testRunId));
+  if (!testRunId) {
+    cliApiKeys = cliApiKeys.filter(isAtLeastOneDayOld);
```

Comment:
The check on `testRunId` means that it will only delete `isAtLeastOneDayOld` only when running the cron job ?

If so, I would just this an explicit script parameter (an argument to the script), instead of relying on a somewhat unclear environment variable injected by github.

Suggestion:
```
node ./scripts/delete-api-keys.js --onlyOlderThan 1d
```